Automated cherry pick of #71681: Add e2e test for file exec

verult opened the pull request:
Cherry pick of #71681 on release-1.11.

#71681: Add e2e test for file exec

```release-note
NONE
```

msau42 committed:
Add e2e test for file exec

verult commented:
/kind bug
/priority important-soon
/assign @msau42 
/sig storage

msau42 commented on test/e2e/storage/volumes.go:
```diff
@@ -119,6 +124,20 @@ var _ = utils.SIGDescribe("Volumes", func() {
+				NFS: &v1.NFSVolumeSource{
+					Server:   serverIP,
+					Path:     "/",
+					ReadOnly: true,
```
By setting readOnly, I think we won't be able to echo the script to exec

verult commented:
This is adding a new test case for a bug found with certain kernel versions.

/assign @foxish

verult commented:
Addressed comments

foxish commented:
@msau42, is this good to go?

verult commented:
Fixed the gofmt error, but I'm not sure where the bazel build error came from. Maybe a flake?

@msau42 btw I changed the test image from Nginx to NginxSlim (since the former isn't defined in 1.11). Also simplified the test function a bit by removing the bits tailored to the new e2e testing framework

msau42 commented:
/lgtm
/approve

verult commented:
@foxish good to go!
